### ISSUE TITLE
Build tag to indicate syscall/js support?

### ISSUE THREAD

**Issue by flimzy** [gopherjs/gopherjs#920]
Would it be reasonable to add a build tag to indicate syscall/js support in GopherJS?

Maybe it's not worth it, as I believe such a tag will only be useful until GopherJS supports Go 1.13, as then the `go1.13` tag could serve the same purpose.

But until then, to retain backward compatibility with GopherJS 1.12 prior to #908, a build tag could come in handy.

Without such a tag, I'll probably wait until GopherJS 1.13 is released before updating my libraries to use the syscall/js package, so that I can be sure to retain backward compatibility. And this isn't a terrible outcome.

**Comment by dmitshur** [comment]
Do you have a suggestion for what the build tag should be called? Should it be something like `gopherjs1.12-3` (to be included in GopherJS 1.12-3 release)?

There’s some precedent for setting a custom build tag as in https://github.com/gopherjs/gopherjs/pull/865, but that was to maintain backwards compatibility with an older Go point release. This is about indicating a version of GopherJS.

**Comment by dmitshur** [comment]
Also, do you know if anyone uses a non-latest point release of GopherJS, or is this more of a theoretical issue?

**Comment by flimzy** [comment]
> Do you have a suggestion for what the build tag should be called?

Not really.

>Should it be something like `gopherjs1.12-3` (to be included in GopherJS 1.12-3 release)?

That seems perfectly reasonable to me.

> Also, do you know if anyone uses a non-latest point release of GopherJS, or is this more of a theoretical issue?

I don't know, except that from personal experience, I often fall behind, just due to not expressly updating my GopherJS version.  I also suppose the fact that people have asked in the past for support to vendor GopherJS would be an indication that some people use, or at least expect to use, historical versions of GopherJS in some contexts.

**Comment by dmitshur** [comment]
`gopherjs1.12-3` won't work:

```
// Tags must be letters, digits, underscores or dots.
// Unlike in Go identifiers, all digits are fine (e.g., "386").
```

*(Source: https://github.com/golang/go/blob/c6f9321b5a46e4af3233417d068c0efb4113d8e7/src/go/build/build.go#L1670-L1671\)*

`gopherjs1.12_3` then?

**Comment by flimzy** [comment]
> `gopherjs1.12_3` then?

Sounds good to me.

**Comment by dmitshur** [comment]
One potential problem with adding a build tag like this is that we'll need to always continue to set it in future versions of GopherJS. If we take it out, that can break some package that has started to depend on it. Something to consider.

**Comment by hajimehoshi** [comment]
The current problem is that Go's `syscall/js` might be changed with breaking compatibility, right? So, would `go1.12` or `go1.13` work? Sorry if I am missing something.

**Comment by flimzy** [comment]
`go1.12` won't work, as there are published versions of gopherjs 1.12 that don't support syscall/js at all. If the API is likely to change before Go 1.13, then waiting probably makes sense.

**Comment by flimzy** [comment]
Now in 2021, I think using `go1.16` is sufficient for this. Closing.
